app/models/mixins/purging_mixin.rb

```diff
@@ -39,10 +39,40 @@
 #     :miq_report_id
 #   end
 #
+# For purge_timer to work, the mixee may consider overriding the following method:
+#
+#   @return [:remain, <Date|>
+#   def purge_mode_and_value
+#      value = ::Settings.reporting.history.keep_reports
+#      mode  = value.number_with_method? ? :date : :remaining
+#      value = value.to_i_with_method.seconds.ago.utc if mode == :date
+#      return mode, value
+#   end
+#
 module PurgingMixin
   extend ActiveSupport::Concern
 
   module ClassMethods
+    # This provides backward support for date only implementations
+    # It is assumed most classes will override to provide `:remaining` support
```

Comment:
If most classes will override, then why not make that the default and only do the rare places for override.

Even so, I think this comment is wrong, and I can't understand it.  Everything supports by date, so why does it say "backward support for date only implementations".  Nearly all of them are date only, right?